BinaryEncoding.md

```diff
@@ -67,18 +67,30 @@ Yes:
 
 * A module contains:
   * a header followed by
-  * a table (sorted by offset) containing, for each section, its type and offset (within the module), followed by
+  * a table (sorted by offset) containing, for each section, its type and
+    offset (within the module), followed by
   * a sequence of sections.
 * A section contains:
   * a header followed by
   * the section contents (specific to the section type)
+* A definitions section contains:
+  * the generic section header
+  * a table containing, for each opcode-space, a standardized string literal
+    type name (where index defines its type), offset (within the section),
+    sorted by offset, followed by
+  * a sequence of opcode tables
+  * An opcode table contains:
+    * a sequence of standardized string literal opcode names, where order
+      determines opcode index
 * A code section contains:
   * the generic section header
-  * a table containing, for each function, it's signature, offset (within the section), sorted by offset, followed by
+  * a table containing, for each function, its signature, offset (within the
+    section), sorted by offset, followed by
   * a sequence of functions
-* A function contains:
-  * a table containing, for each type, how many locals are indexed by the function body of that type
-  * the serialized AST
+  * A function contains:
+    * a table containing, for each type, how many locals are indexed by the
+      function body of that type
```

Comment:
What if a function doesn't use some types? Do we need a bunch of zero-sized entries for all SIMD types?